src/components/PopUpWithConfirmation.js

```diff
@@ -0,0 +1,33 @@
+import PopUp from "./PopUp";
+
+export default class PopUpWithConfirmation extends PopUp {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.